admin/services/data-access/azure-blob.data.service.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const azureStorage = require('azure-storage')
+const config = require('../../config')
+let blobService
+
+if (config.AZURE_STORAGE_CONNECTION_STRING) {
+  blobService = azureStorage.createBlobService()
+}
+
+const service = {
+  getBlobText: async (containerName, blobName) => {
+    return new Promise((resolve, reject) => {
+      if (!blobService) {
+        reject(new Error('Azure Storage Connection String required'))
+      }
+      blobService.getBlobToText(containerName, blobName, (error, text) => {
+        if (error) reject(error)
```

Comment:
Should be `return reject` or it will drop through into the `return resolve` line immediately after.